src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java

```diff
@@ -65,16 +69,20 @@ private SmithyProject(List<Path> imports, List<File> smithyFiles, List<File> ext
      * @return either an error, or a loaded project
      */
     public Either<Exception, SmithyProject> recompile(File file) {
+
         // We aggressively re-build the model with only existing files
         // it's simpler than trying to manage which file was added/removed/closed etc.
-        List<File> newFiles = new ArrayList<File>();
-        newFiles.addAll(onlyExistingFiles(this.smithyFiles));
+        List<File> newFiles = new ArrayList<File>(onlyExistingFiles(this.smithyFiles));
 
         if (file.isFile()) {
             newFiles.add(file);
         }
 
-        return load(this.imports, this.smithyFiles, this.externalJars, this.root);
```

Comment:
This was broken (notice the `this.smithyFiles` even though we need to use `newFiles`)

Found with the test!